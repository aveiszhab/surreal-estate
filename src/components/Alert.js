import React from "react";
import "../styles/Alert.css";
import PropTypes from "prop-types";

const Alert = ({ message, success }) => {
  if (!message) return null;
  return (
    <div className={`alert ${success ? "success" : "error"}`}>{message}</div>
  );
};

Alert.propTypes = {
  message: PropTypes.string,
  success: PropTypes.bool,
};
Alert.defaultProps = {
  message: "",
  success: false,
};
export default Alert;
