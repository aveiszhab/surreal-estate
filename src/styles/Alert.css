.alert{
    font-weight: 600;
    text-align: center;
}

.error{
    background-color: rgb(187, 83, 83);
    border: 1px solid rgb(1, 44, 1)
}

.success{
    background-color: green;
    border: 1px solid rgb(1, 44, 1);
}