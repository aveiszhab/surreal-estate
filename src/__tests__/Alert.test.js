import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert", () => {
  it("renders correctly to match the Alert snapshot", () => {
    const { asFragment } = render(<Alert />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("displays an error message", () => {
    const { getByText, asFragment } = render(<Alert message="Error!" />);
    expect(getByText(/error/i).textContent).toContain("Error");
    expect(asFragment()).toMatchSnapshot();
  });
  it("displays a success message", () => {
    const { getByText, asFragment } = render(
      <Alert message="Success!!!" success />
    );
    expect(getByText(/success/i).textContent).toContain("Success");
    expect(asFragment()).toMatchSnapshot();
  });
  it("does not render component if there is no message", () => {
    const { asFragment } = render(<Alert message="" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
