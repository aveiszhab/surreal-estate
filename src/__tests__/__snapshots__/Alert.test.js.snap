// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert displays a success message 1`] = `
<DocumentFragment>
  <div
    class="alert success"
  >
    Success!!!
  </div>
</DocumentFragment>
`;

exports[`Alert displays an error message 1`] = `
<DocumentFragment>
  <div
    class="alert error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert does not render component if there is no message 1`] = `<DocumentFragment />`;

exports[`Alert renders correctly to match the Alert snapshot 1`] = `<DocumentFragment />`;
